### PR TITLE
[WIP] Sponge Port

### PR DESCRIPTION
**Pull request by meyerzinn**
Things to do:
- Test GUIs when that's implemented.
- Finish implementing signs.
- Overall tuneups.

### CODE REVIEW

File: sponge/src/main/java/net/buycraft/plugin/sponge/BuycraftPlugin.java [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+            listingUpdateTask.run();
+            Sponge.getScheduler().createTaskBuilder().delayTicks(20 * 60 * 20).intervalTicks(20 * 60 * 20).execute(listingUpdateTask).async()
+                    .submit(this);
+        }
+
+        Sponge.getEventManager().registerListeners(this, new BuycraftListener(this));
+
+        /*
+        *TODO: Commands.
+         */
```
Comment (minecrafter) [inline sponge/src/main/java/net/buycraft/plugin/sponge/BuycraftPlugin.java:138]: Leftover comments

Comment (meyerzinn) [inline sponge/src/main/java/net/buycraft/plugin/sponge/BuycraftPlugin.java:138]: It’s still a WIP but thanks.

File: sponge/src/main/java/net/buycraft/plugin/sponge/BuycraftPlugin.java [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+                .description(Text.of("Lists all Buycraft packages and their prices."))
+                .executor(new ListPackagesCmd(this))
+                .build();
+        return CommandSpec.builder()
+                .description(Text.of("Main command for the Buycraft plugin."))
+                .child(report, "report")
+                .child(list, "list", "packages")
+                .child(secret, "secret")
+                .child(refresh, "refresh")
+                .executor(new GuiCmd(this))
```
Comment (minecrafter) [inline sponge/src/main/java/net/buycraft/plugin/sponge/BuycraftPlugin.java]: Looks like left-over stuff. GUI isn't supported. Even then, the command is defined separately.

Comment (meyerzinn) [inline sponge/src/main/java/net/buycraft/plugin/sponge/BuycraftPlugin.java]: Yeah, just committed a fix. Will add.

File: sponge/src/main/java/net/buycraft/plugin/sponge/SpongeBuycraftPlatform.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    @Override public void executeBlocking(Runnable runnable) {
+        Sponge.getScheduler().createTaskBuilder().execute(runnable).submit(plugin);
+    }
+
+    @Override public void executeBlockingLater(Runnable runnable, long time, TimeUnit unit) {
+        Sponge.getScheduler().createTaskBuilder().execute(runnable).delay(time, unit).submit(plugin);
+    }
+
+    @Override public boolean isPlayerOnline(QueuedPlayer player) {
+        String uuidwd = player.getUuid().replaceFirst("(\\p{XDigit}{8})(\\p{XDigit}{4})(\\p{XDigit}{4})(\\p{XDigit}{4})(\\p{XDigit}+)",
```
Comment (minecrafter) [inline sponge/src/main/java/net/buycraft/plugin/sponge/SpongeBuycraftPlatform.java:59]: Does not take into account that `player.getUuid()` may be `null` - look at how the Bukkit port handles this.

File: sponge/src/main/java/net/buycraft/plugin/sponge/signs/purchases/RecentPurchaseSignListener.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+            event.getText().lines().set(i, Text.EMPTY);
+        }
+
+        plugin.getPlatform().executeAsync(new SignUpdater(plugin));
+    }
+
+    @Listener
+    public void onBlockBreak(ChangeBlockEvent.Break event) {
+        // This is what happens when you code without sleep.
+        // TODO Someone should check if this actually works .-.
```
Comment (minecrafter) [inline sponge/src/main/java/net/buycraft/plugin/sponge/signs/purchases/RecentPurchaseSignListener.java:76]: More comment stuff

File: sponge/src/main/java/net/buycraft/plugin/sponge/signs/purchases/RecentPurchaseSignPosition.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Value;
+import org.spongepowered.api.world.Location;
+import org.spongepowered.api.world.World;
+
+@Value
+@AllArgsConstructor
+public class RecentPurchaseSignPosition {
+
+    private final Location<World> location;
```
Comment (minecrafter) [inline sponge/src/main/java/net/buycraft/plugin/sponge/signs/purchases/RecentPurchaseSignPosition.java:12]: Is this safe to serialize with Gson?

File: sponge/src/main/java/net/buycraft/plugin/sponge/signs/purchases/RecentPurchaseSignListener.java (hunk 4 above)
Comment (meyerzinn) [inline sponge/src/main/java/net/buycraft/plugin/sponge/signs/purchases/RecentPurchaseSignListener.java:76]: The Sign section is under heavy development.

File: sponge/src/main/java/net/buycraft/plugin/sponge/signs/purchases/RecentPurchaseSignPosition.java (hunk 5 above)
Comment (meyerzinn) [inline sponge/src/main/java/net/buycraft/plugin/sponge/signs/purchases/RecentPurchaseSignPosition.java:12]: No clue. Was going to implement a different method anyways, using Sponge’s Serializable.

File: sponge/src/main/java/net/buycraft/plugin/sponge/tasks/BuyNowSignUpdater.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+/**
+ * Created by meyerzinn on 2/14/16.
+ */
+@AllArgsConstructor
+public class BuyNowSignUpdater implements Runnable{
+
+    private final BuycraftPlugin plugin;
+
+    @Override public void run() {
+
```
Comment (minecrafter) [inline sponge/src/main/java/net/buycraft/plugin/sponge/tasks/BuyNowSignUpdater.java:15]: This class needs to be implemented.

**Comment by minecrafter** [comment]
Need you to do this as well:
- Remove your `.gitignore` and `.travis.yml`.
- Readd `README.md`
- Squash your commits

**Comment by minecrafter** [comment]
I've decided to move this into a [separate branch](https://github.com/BuycraftPlugin/BuycraftX/tree/sponge-port) to give it better attention.

I'll be working on this a bit more and it'll be merged in soon.

**Pull request closed without merging**